Hermes/appscale/hermes/stats/stats_app.py

```diff
@@ -58,357 +53,155 @@ class HandlerInfo(object):
+  node, processes and proxies stats (on master node only).
+  If this node is slave, it creates stub handlers for cluster stats routes.
 
-  def configure(self):
-    """ Builds publishers list and routes for stats API
-        according to configurations
-    """
-    # Every single node produces node stats
-    self._init_local_node_stats_publisher()
+  Args:
+    is_master: a boolean indicating whether this node is master
+  Returns:
+    a list of route-handler tuples
+  """
+  if is_master:
+    # Only master node provides cluster stats
+    cluster_node_stats_handler = HandlerInfo(
+      handler_class=CurrentClusterStatsHandler,
+      init_kwargs={'source': ClusterNodesStatsSource(),
+                   'default_include_lists': DEFAULT_INCLUDE_LISTS}
+    )
+    cluster_processes_stats_handler = HandlerInfo(
+      handler_class=CurrentClusterStatsHandler,
+      init_kwargs={'source': ClusterProcessesStatsSource(),
+                   'default_include_lists': DEFAULT_INCLUDE_LISTS}
+    )
+    cluster_proxies_stats_handler = HandlerInfo(
+      handler_class=CurrentClusterStatsHandler,
+      init_kwargs={'source': ClusterProxiesStatsSource(),
+                   'default_include_lists': DEFAULT_INCLUDE_LISTS}
+    )
+  else:
+    # Stub handler for slave nodes
+    cluster_stub_handler = HandlerInfo(
+      handler_class=Respond404Handler,
+      init_kwargs={'reason': 'Only master node provides cluster stats'}
+    )
+    cluster_node_stats_handler = cluster_stub_handler
+    cluster_processes_stats_handler = cluster_stub_handler
+    cluster_proxies_stats_handler = cluster_stub_handler
 
-    if self._track_processes:
-      # Processes stats are optional
-      self._init_local_processes_stats_publisher()
-    else:
-      self._stub_processes_stats_routes()
+  routes = {
+    '/stats/cluster/nodes': cluster_node_stats_handler,
+    '/stats/cluster/processes': cluster_processes_stats_handler,
+    '/stats/cluster/proxies': cluster_proxies_stats_handler,
+  }
+  return [
+    (route, handler.handler_class, handler.init_kwargs)
+    for route, handler in routes.iteritems()
+  ]
 
-    if self._is_lb:
-      # Load balancer node also provides proxies stats
-      self._init_local_proxies_stats_publisher()
-    else:
-      self._stub_proxies_stats_routes()
 
-    if self._is_master:
-      # Master collects stats from all nodes and provides API for access
-      self._init_cluster_node_stats_publisher()
-      if self._track_processes:
-        self._init_cluster_processes_stats_publisher()
-      self._init_cluster_proxies_stats_publisher()
-    else:
-      self._stub_cluster_stats_routes()
+def _configure_profiling(stats_source, profiler, interval):
 
-  def start_publishers(self):
-    """ Starts tornado periodic tasks for each of configured stats publishers
+  def write_stats_callback(future_stats):
+    """ Is called when cluster stats were collected and are ready to be written
```

Comment:
Maybe change the docstring to indicate what the method does and not when it is called to be uniform with the rest of the doc strings. Maybe "Writes the profiler stats when...."